uaclient/contract_data_types.py

```diff
@@ -0,0 +1,602 @@
+        Field("createdBy", StringDataValue, False),
+        Field("resourceEntitlements", data_list(Entitlement), False),
+        Field("specificResourceEntitlements", data_list(Entitlement), False),
+        Field("effectiveFrom", StringDataValue, False),
+        Field("effectiveTo", StringDataValue, False),
+        Field("products", data_list(StringDataValue), False),
+        Field("origin", StringDataValue, False),
+        Field("activeMachines", ActiveMachine, False),
+        Field("items", data_list(ContractItem), False),
+        Field("renewals", data_list(Renewal), False),
+        Field("externalAssetIDs", ExternalID, False),
+        Field("allowances", data_list(AllowanceInfo), False),
+    ]
+
+    def __init__(
+        self,
+        name: Optional[str],
+        id: Optional[str],
+        createdAt: Optional[str],
+        createdBy: Optional[str],
+        resourceEntitlements: Optional[List[Entitlement]],
+        specificResourceEntitlements: Optional[List[Entitlement]],
+        effectiveFrom: Optional[str],
+        effectiveTo: Optional[str],
+        products: Optional[List[str]],
+        origin: Optional[str],
+        activeMachines: Optional[ActiveMachine],
+        items: Optional[List[ContractItem]],
+        renewals: Optional[List[Renewal]],
+        externalAssetIDs: Optional[ExternalID],
+        allowances: Optional[List[AllowanceInfo]],
+    ):
+        self.name = name
+        self.id = id
+        self.createdAt = createdAt
+        self.createdBy = createdBy
+        self.resourceEntitlements = resourceEntitlements
+        self.specificResourceEntitlements = specificResourceEntitlements
+        self.effectiveFrom = effectiveFrom
+        self.effectiveTo = effectiveTo
+        self.products = products
+        self.origin = origin
+        self.activeMachines = activeMachines
+        self.items = items
+        self.renewals = renewals
+        self.externalAssetIDs = externalAssetIDs
+        self.allowances = allowances
+
+
+class MachineTokenInfo(DataObject):
+    fields = [
+        Field("machineId", StringDataValue, False),
+        Field("accountInfo", AccountInfo, False),
+        Field("contractInfo", ContractInfo, False),
+        Field("expires", StringDataValue, False),
+    ]
+
+    def __init__(
+        self,
+        machineId: Optional[str],
+        accountInfo: Optional[AccountInfo],
+        contractInfo: Optional[ContractInfo],
+        expires: Optional[str],
+    ):
+        self.machineId = machineId
+        self.accountInfo = accountInfo
+        self.contractInfo = contractInfo
+        self.expires = expires
+
+
+class PublicMachineTokenData(DataObject):
+    fields = [
+        Field("activityInfo", ActivityInfo, False),
+        Field("machineTokenInfo", MachineTokenInfo, False),
+        Field("availableResources", data_list(AvailableResource), False),
+        Field("resourceTokens", data_list(ResourceToken), False),
```

Comment:
@renanrodrigo @lucasmoura @CalvoM what if we just drop `resourceTokens` from the public file?

Reply:
+1